src/components/astra-editor/index.ts

```diff
@@ -0,0 +1,351 @@
+import { acceptCompletion, autocompletion, closeBracketsKeymap, completionStatus, startCompletion } from '@codemirror/autocomplete'
+import { defaultKeymap, history, historyKeymap, indentLess, indentMore, insertNewline } from '@codemirror/commands'
+import { defaultHighlightStyle, foldKeymap, indentOnInput, indentUnit, syntaxHighlighting } from '@codemirror/language'
+import { Compartment, Prec, StateEffect, type Extension } from '@codemirror/state'
+import {
+  dropCursor,
+  highlightActiveLine,
+  highlightActiveLineGutter,
+  keymap,
+  lineNumbers,
+  placeholder,
+  ViewUpdate,
+  type KeyBinding,
+} from '@codemirror/view'
+import { EditorView } from 'codemirror'
+import { LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+import { getPredefineTheme } from './theme'
+
+@customElement('astra-editor')
+export class AstraEditor extends LitElement {
+  protected container: HTMLDivElement
+  protected editor?: EditorView
+  protected extensions: { name: string; comp: Compartment; ext: Extension }[] = []
+
+  protected _value: string = ''
+  protected _wrap: boolean = false
+  protected _color: 'dark' | 'light' = 'light'
+  protected _theme: string = 'moondust'
+  protected _placeholder: string = ''
+
+  constructor() {
+    super()
+
+    const container = document.createElement('div')
+    container.style.height = '100%'
+
+    this.attachShadow({ mode: 'open' })
+    this.shadowRoot?.append(container)
```

Comment:
When using litelement you don't typically need to create your own shadowRoot and manipulate the DOM manually.  I'm not sure if there is something special about this editor that we do need/want to here?
Might be able to use the `styles` field instead of manually crafting `<style />` tags?

https://lit.dev/docs/components/overview/

Reply:
I also tried `styles`, but the way I use `render()` does not seem to support `static styles`